lumicks/pylake/kymo.py

```diff
@@ -618,18 +643,24 @@ def estimate_bead_edges(
                 f"functionality."
             )
 
-        return (
-            find_beads(
-                self.get_image(channel).sum(axis=1),
-                bead_diameter_pixels=bead_diameter / self.pixelsize_um[0],
-                plot=plot,
-                threshold_percentile=threshold_percentile,
-                **kwargs,
+        image = self.get_image(channel)
+        shared_parameters = {
+            "bead_diameter_pixels": bead_diameter / self.pixelsize_um[0],
+            "plot": plot,
+        } | kwargs
+
+        if algorithm == "brightness":
+            return (
+                np.array(find_beads(image.sum(axis=1), **shared_parameters)) * self.pixelsize_um[0]
+            )
+        elif algorithm == "template":
+            return np.array(find_beads_template(image, **shared_parameters)) * self.pixelsize_um[0]
+        else:
+            raise ValueError(
+                f'Unrecognized algorithm {algorithm} selected. Choose "brightness" or "template"'
```

Comment:
If you go with my suggestion for both private functions to take the full kymo image you could do something like
```
if algorithm not in ("brightness", "template"):
    raise ValueError(...)
finder = find_beads if algorithm == "brightness" else find_beads_template
return np.array(finder(image, **shared_parameters)) * self.pixelsize_um[0]
```